app/models/null_service_provider.rb

```diff
@@ -28,4 +28,20 @@ def fingerprint
   def ssl_cert
     nil
   end
+
+  def logo
+    nil
+  end
+
+  def friendly_name
+    'NullServiceProvider'
```

Comment:
Maybe just leave it `nil` to avoid confusion? We don't need a string. I wasn't trying to name it the same as the class.